### PR TITLE
fix(ENG-2114): Fix port calculation overflow for high ticket numbers

### PR DESCRIPTION
**Pull request by K-Mistele**
## Summary
- Fixed critical bug in port calculation that caused "invalid port number" errors for tickets ≥ 1555
- Ensures all generated ports stay within valid range (1-65535)
- Implements smarter fallback strategies for Vite ports

## Problem
When running `make codelayer-dev TICKET=ENG-1937`, the command would fail with:
```
invalid port number: "http://localhost:111937"
```

The port fallback logic in `hack/port-utils.sh` was using string concatenation instead of arithmetic, causing it to generate invalid ports > 65535. For example, when Vite base port 11937 was unavailable, it would concatenate "1" + "11937" = "111937" instead of finding a valid alternative.

## Solution
1. **Added port validation** in `find_available_port()` to skip any ports exceeding 65535
2. **Rewrote `find_available_vite_port()`** with a smarter fallback strategy:
   - First tries base port (ticket + 10000) if valid
   - Falls back to 30000-65000 range for high tickets  
   - Secondary fallback to 40000-50000 range
   - All ports guaranteed to be within valid range

## Test plan
- [x] Tested with ENG-1937: Correctly allocates daemon port 1937, Vite port 11937
- [x] Tested with port conflicts: Properly falls back to valid alternatives
- [x] Validated edge cases: ENG-1555 (boundary), ENG-9999 (high ticket)
- [x] All generated ports verified to be ≤ 65535

----

> [!IMPORTANT]
> Fixes port calculation overflow for high ticket numbers in `hack/port-utils.sh`, ensuring valid port ranges and smarter fallback strategies.
> 
>   - **Behavior**:
>     - Fixes port calculation overflow in `hack/port-utils.sh` for tickets ≥ 1555, ensuring ports stay within 1-65535.
>     - Implements smarter fallback strategies in `find_available_vite_port()` for Vite ports.
>   - **Makefile**:
>     - Updates `wui-ticket` target to use `bun run` instead of `bun tauri dev`.
>   - **Testing**:
>     - Validated with ENG-1937, port conflicts, and edge cases like ENG-1555 and ENG-9999.
>     - Ensures all generated ports are ≤ 65535.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=humanlayer%2Fhumanlayer&utm_source=github&utm_medium=referral)<sup> for 6e2f2b1a67795b6f1a808becb8e5aa997d1f5dcf. You can [customize](https://app.ellipsis.dev/humanlayer/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Review by dexhorthy — APPROVED** [review]

File: .claude/settings.local.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
```
Comment (dexhorthy) [inline .claude/settings.local.json:1]: we should commit `settings.json` but not `settings.local.json`

File: humanlayer-wui/src/components/Breadcrumbs.tsx [hunk 2]
```diff
@@ -81,9 +89,36 @@ export function Breadcrumbs() {
...
+                        e.preventDefault()
+                        e.stopPropagation()
+                        // Stop immediate propagation to prevent any other handlers
+                        e.nativeEvent.stopImmediatePropagation()
+                        // Reset edit value to original title
+                        setEditValue(session?.title || session?.summary || '')
+                        // Exit editing mode without saving
+                        setIsEditingTitle(false)
+                        // Just blur the input, don't focus anything else
+                        ;(e.target as HTMLInputElement).blur()
```
Comment (dexhorthy) [inline humanlayer-wui/src/components/Breadcrumbs.tsx:114]: damn you found an ASI thing nice

**Comment by K-Mistele** [comment]
Man, this is the second time that this has happened today, but it looks like there's some changes from another PR in this branch, so I'm going to create a new PR.
